microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc

```diff
@@ -0,0 +1,2559 @@
+| Read Timeout Time unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  Time unit for the *Read Timeout* field. |  SECONDS |
+|===
+
+==== Connection Types
+[[Outlook365Config_OauthAuthorizationCode]]
+===== OAuth v2.0 Authorization Code
+
+The OAuth 2.0 authorization code grant connection type is used in apps that are installed on a device to gain access to protected resources.
+
+
+====== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Proxy configuration a| <<ProxyConfiguration>> |  Configures a proxy for outbound connections. |  |
+| Response Type a| String |  Specifies the method to use to send the resulting token back to your app. This must be `query`. |  query |
+| TLS configuration a| <<Tls>> |  Configures TLS. If using the HTTPS protocol, you must configure TLS.  |  |
+| Reconnection a| <<Reconnection>> |  Configures a reconnection strategy to use when a connector operation fails to connect to an external server. |  |
+| Consumer Key a| String | OAuth consumer key, as registered with the service provider. |  | x
+| Consumer Secret a| String |  OAuth consumer secret, as registered with the service provider. |  | x
+| Authorization Url a| String |  URL of the service provider's authorization endpoint. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/authorize |
+| Access Token Url a| String |  URL of the service provider's access token endpoint. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token |
+| Scopes a| String |  OAuth scopes to request during the OAuth dance. This value defaults to the scopes in the annotation. |  |
+| Resource Owner Id a| String |  Resource owner ID to use with the authorization code grant type. |  |
+| Before a| String |  Name of the flow to execute immediately before starting the OAuth dance. |  |
+| After a| String |  Name of the flow to execute immediately after receiving an accessToken. |  |
+| Listener Config a| String |  Configuration for the HTTP listener that listens for requests on the access token callback endpoint. |  | x
+| Callback Path a| String |  Path of the access token callback endpoint. |  | x
+| Authorize Path a| String |  Path of the local HTTP endpoint that triggers the OAuth dance. |  | x
+| External Callback Url a| String |  URL that the OAuth provider uses to access the callback endpoint if the endpoint is behind a proxy or accessed through an indirect URL. |  |
+| Object Store a| String |  Configures the object store that stores data for each resource owner. If not configured, Mule uses the default object store. |  |
+|===
+[[Outlook365Config_OauthClientCredentials]]
+===== OAuth v2.0 Client Credentials
+
+The OAuth 2.0 client credentials grant accesses web-hosted resources by using the identity of the application.
```

Comment:
The OAuth 2.0 client credentials grant access to web-hosted resources by using the identity of the application.